include/revng/Model/Binary.h

```diff
@@ -614,6 +614,7 @@ class model::Binary {
   bool verify(bool Assert) const debug_function;
   bool verify(VerifyHelper &VH) const;
   void dump() const debug_function;
+  void dumpOnFile(const char *FileName) const debug_function;
```

Comment:
```suggestion
  template<typename T>
  void dump(T &Stream) const;
  void dump() const debug_function { dump(dbg); }
```

Reply:
Can I use this inside `gdb` to dump the model at arbitrary points on a file for later inspection?

Reply:
Mmmh, we have a `dumpModule` function for `llvm::Module` for `gdb`.
Not sure, I don't want this on every possible part of the model, but maybe we can have it on the root.

Maybe we can have a `std::string toString()` method and then a global function `writeToFile(StringRef Path, StringRef What)`.
```
(gdb) p writeToFile(Model.toString(), "/tmp/asd.yml");
```

What do you think?

Reply:
That would be nice

Reply:
Where is the best place to put `writeToFile()`?

Reply:
`include/revng/Support/Debug.h` for now.

Reply:
I noticed that the order you suggested was different in the gdb example and in the description of the to-be function signature. I went for the former (`writeToFile(StringRef What, StringRef Path)`).